src/common/state/redis.cc

```diff
@@ -77,52 +78,77 @@ extern int usleep(useconds_t usec);
   do {                                                                         \
   } while (0)
 
-DBHandle *db_connect(const char *db_address,
-                     int db_port,
-                     const char *client_type,
-                     const char *node_ip_address,
-                     int num_args,
-                     const char **args) {
-  /* Check that the number of args is even. These args will be passed to the
-   * RAY.CONNECT Redis command, which takes arguments in pairs. */
-  if (num_args % 2 != 0) {
-    LOG_FATAL("The number of extra args must be divisible by two.");
+redisAsyncContext *get_redis_context(DBHandle *db, UniqueID id) {
+  UniqueIDHasher index;
+  return db->contexts[index(id) % db->contexts.size()];
```

Comment:
We may want to document the fact that if we change this hash function, we also need to change `PyObjectID_redis_shard_hash` in `common_extension.cc`